test/archethic/contracts/interpreter/scope_test.exs

```diff
@@ -5,55 +5,589 @@ defmodule Archethic.Contracts.Interpreter.ScopeTest do
 
   doctest Scope
 
-  test "global variables" do
-    Scope.init(%{"var1" => %{"prop" => 1}})
-    Scope.write_at([], "var2", 2)
-    assert Scope.read_global(["var1", "prop"]) == 1
-    assert Scope.read_global(["var2"]) == 2
+  describe "init/1" do
+    test "should instantiate Scope with empty context list and global vars" do
+      constants = %{"var1" => %{"prop" => 1}, "var2" => 4}
+      Scope.init(constants)
+      assert %{"context_list" => [], "var1" => %{"prop" => 1}, "var2" => 4} == Process.get(:scope)
+    end
   end
 
-  test "write_at/3" do
-    Scope.init()
+  describe "create_context/0" do
+    test "create_context/0 should create empty context to existing scope" do
```

Comment:
In test title you don't need to repeat the describe title since the log test concatenate both